File: Leetcode/11-container-with-most-water.cpp
class Solution {
    public:
        int maxArea(vector<int>& height) {
            int max_area = INT32_MIN;
            int width;
            for (int left = 0, right = height.size() - 1; left < right;)
            {
                width = right - left;
                max_area = max(max_area, min(height[left], height[right]) * width);
                
                height[left] <= height[right] ? left++ : right--;
            }
            return max_area;   
        }
    };

File: Leetcode/11-Container-With-Most-Water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        maxArea = float('-inf')
        left = 0
        right = len(height) - 1
        
        while left < right:
            width = right - left
            if height[left] <= height[right]:
                maxArea = max(maxArea, height[left] * width)
                left += 1
            else:
                maxArea = max(maxArea, height[right] * width)
                right -= 1
        
        return maxArea